cloudnet-modules/cloudnet-database-mongodb/src/main/java/de/dytanic/cloudnet/ext/database/mongodb/MongoDatabase.java

```
@@ -0,0 +1,334 @@
+package de.dytanic.cloudnet.ext.database.mongodb;
+
+import com.google.common.base.Preconditions;
+import com.mongodb.BasicDBObject;
+import com.mongodb.MongoCommandException;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import de.dytanic.cloudnet.common.concurrent.ITask;
+import de.dytanic.cloudnet.common.concurrent.ListenableTask;
+import de.dytanic.cloudnet.common.document.gson.JsonDocument;
+import de.dytanic.cloudnet.driver.CloudNetDriver;
+import de.dytanic.cloudnet.driver.database.Database;
+import de.dytanic.cloudnet.ext.database.mongodb.util.DocumentUtil;
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import lombok.Getter;
+import org.bson.Document;
+import org.jetbrains.annotations.NotNull;
+
+@Getter
+public class MongoDatabase implements Database {
+
+    private static final String COLLECTION_KEY = "==key==";
+
+    protected final MongoDatabaseProvider mongoDatabaseProvider;
+    protected final String name;
+    protected final ExecutorService executorService;
+    private final MongoCollection<Document> collection;
+
+    public MongoDatabase(MongoDatabaseProvider mongoDatabaseProvider, String name, ExecutorService executorService) {
+        Preconditions.checkNotNull(mongoDatabaseProvider);
+        Preconditions.checkNotNull(name);
+        Preconditions.checkNotNull(executorService);
+        this.executorService = executorService;
+        this.mongoDatabaseProvider = mongoDatabaseProvider;
+        this.name = name;
+        try {
+            mongoDatabaseProvider.getMongoDatabase().createCollection(name);
+        } catch (MongoCommandException e) {
```

Don't ignore exceptions, handle it. 
What if when!!